### DISCUSSION TITLE
How to concurrently use AA in wireless mode and mobile data

### DISCUSSION THREAD

**Discussion by frank-cerebras** [opencardev/crankshaft#535] [Q&A]
Hi everyone, I have a question on how to concurrently use AA in wireless mode and mobile data.

I own a Galaxy Z Flip 3.

I followed [these](https://github.com/opencardev/crankshaft/wiki/Android-Auto-Wifi) steps to have crankshaft run in hotspot mode, and my phone was able to successfully connect to crankshaft's wifi.

When I connect to crankshaft's wifi, it says that the wifi "might not have internet connection", which is an expected message. I then click "connect anyway", which allows me to forcefully connect to the wifi. At this point, the RPi is successfully displaying AA, but it is not connected to mobile data (because I forcefully connected to the wifi thereby turning off mobile data).

Note: I have already enabled "Switch to mobile data" but it seems like "connect anyway" takes precedence over this.

There IS a workaround to get the mobile data working concurrently, however. When I go into "Switch to mobile data" setting, there is a list of "Network exceptions" that states: `Your phone will remain connected to these networks, even if the connection is unstable`. Earlier when I forcefully connected to the wifi, it added an entry in this list. The fix is to remove the entry from the list. This allows the phone to turn on mobile data while being connected to the wifi. At this point, the RPi is displaying AA and I am able to access mobile data (map search, music, etc).

However, this solution is temporary because by removing the entry from the list, the phone prompts me the "wifi might not have internet connection" message when I disconnect and reconnect the wifi. This means that I need to click "connect anyway" and run the same process all over again.

I've enabled "mobile data always on" in developer's settings, but it seems like the "Network exceptions" take precedence over that setting. I feel like there is a way to work around this issue, so I was wondering if anyone else has run into similar issues.

Any response is appreciated!

**Comment by matt2005** [comment]
Have you tried setting you phone to be a personal hotspot and then connect crankshaft to it as a client?

**Reply by joharri9** [reply]
I believe that [those](https://github.com/opencardev/crankshaft/wiki/Android-Auto-Wifi) steps are old documentation.

I have wireless working on Alpha-7.4 2021-09-22 with a Google Pixel 2.  This is a fresh install.  I have changed NOTHING from the defaults after the install.  (Ok, I do change one thing, see below.)  If you have messed with all the different hotspot options, I would suggest a reinstall, I don't really know the proper defaults.

Note: you must use an external USB Bluetooth on the r-pi 3b+, the built-in one is flakey.  So plug one in, and in the CS HW menu, set Bluetooth to "external".  

Then I started Android Auto on my phone, went to the settings "hamburger" and "+ connect a car", "connect using wireless Android Auto".  

Now, before clicking "Connect using Bluetooth", on the CS in the menu "BT / WiFi", click "Remove all paired devices now!".  The CS will go into pairing mode.  Now back on the cell phone, click "Connect using Bluetooth".

It is a little slow, generally, it works.  There is a lot of "studdering" in the audio.  In the CS menu "Audio", set the "Output backend" to "RT audio".  This helps, but does not eliminate the studdering.

**Comment by NobleBest1** [comment]
The only way I have found to make the ‘split horizon (I.E. both WIFI and Mobile Data working at the same time) is to force the applications used within AA to use Mobile Data at all times.
This can be done by;
1.	Start the Settings app.
2.	Tap "Wireless & networks" and then tap "Data usage."
3.	Tap "Network access."
4.	In the list of apps, enter the checkbox for any apps you want working within AA
